lib/App/DuckPAN/Query.pm

```diff
@@ -59,74 +61,96 @@ sub _default { warn "Unhandled event - $_[ARG0]" }
 
 # Event that prints the prompt and waits for input.
 sub _get_user_input {
-	$_[HEAP]{console}->get('Query: ');
+    $_[HEAP]{console}->get('Query: ');
 }
 
 # Event that processes the query
 sub _run_query {
-	my ($k, $h, $query) = @_[KERNEL, HEAP, ARG0];
-
-	my ($app, $blocks) = @$h{qw{app blocks}};
-	Encode::_utf8_on($query);
-
-	try {
-	    my $request = DDG::Request->new(
-	        query_raw => $query,
-	        location => test_location_by_env(),
-	        language => test_language_by_env(),
-	    );
-	    my $hit;
-	    # Iterate through the IAs passing each the query request
-	    for my $b (@$blocks) {
-	        for ($b->request($request)) {
-	            $hit = 1;
-	            $app->emit_info('---', p($_, colored => $app->colors), '---');
-	        }
-	    }
-	    unless ($hit) {
-	        $app->emit_info('Sorry, no hit on your instant answer')
-	    }
-	}
-	catch {
-	    my $error = $_;
-	    if ($error =~ m/Malformed UTF-8 character/) {
-	        $app->emit_info('You got a malformed utf8 error message. Normally' .
-	            ' it means that you tried to enter a special character at the query' .
-	            ' prompt but your interface is not properly configured for utf8.' .
-	            ' Please check the documentation for your terminal, ssh client' .
-	            ' or other client used to execute duckpan.'
-	        );
-	    }
-	    $app->emit_info("Caught error: $error");
-	};
-
-	# Enqueue input event
-	$k->yield('_get_user_input');
+    my ($k, $h, $query, $args) = @_[KERNEL, HEAP, ARG0, ARG2];
+
+    my ($app, $blocks) = @$h{qw{app blocks}};
+    Encode::_utf8_on($query);
+
+    my $repo = $app->get_ia_type;
+
+    try {
+        unless ($repo->{name} eq "Fathead") {
+            my $request = DDG::Request->new(
+                query_raw => $query,
+                location => test_location_by_env(),
+                language => test_language_by_env(),
+            );
+            my $hit;
+            # Iterate through the IAs passing each the query request
+            for my $b (@$blocks) {
+                for ($b->request($request)) {
+                    $hit = 1;
+                    $app->emit_info('---', p($_, colored => $app->colors), '---');
+                }
+            }
+            unless ($hit) {
+                $app->emit_info('Sorry, no Instant Answers returned a result')
+            }
+        }
+        else {
+
+            my $output_txt = path( $app->fathead_output() );
+
+            if ($output_txt->exists) {
+                my $result = App::DuckPAN::Fathead->search_output($query, $output_txt);
+                if ($result){
+                    $app->emit_info('---', "Match found: $output_txt", p($result, colored => $app->colors), '---');
+                }
+                else {
+                    $app->emit_info('Sorry, no matches found in output.txt');
+                }
+            }
+            else {
+                $app->emit_info('Sorry, no output.txt file was not found');
```

Comment:
We should probably not be continuing, is there anything useful we can do without `output_txt`?